src/include/daos_fs.h

```diff
@@ -814,6 +814,24 @@ dfs_access(dfs_t *dfs, dfs_obj_t *parent, const char *name, int mask);
 int
 dfs_chmod(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode);
 
+/**
+ * Change owner and group. Symlinks are dereferenced. Since uid and gid are not enforced at the DFS
```

Comment:
This implies that the equivalent of `lchown` has be done with `dfs_osetattr`, and simlar for `dfs_sys_chown`. Interestingly, mpifileutils uses `chmod` (not `fchmodat`), but `lchown`. For now, `lchown/chown` is ignored for DFS, but I imagine we'd want to eventually update datamover tools to use these. So how do you think `lchown` should be handled? Through `dfs_osetattr`?

Reply:
Assuming "lchown" means link I should add that in posix chown dereferences links, and furthermore link permissions are never used, so all links are mode 777 and cannot be changed.

Reply:
They are mode 777, but I think they still have uid/gid, right? E.g.:
```
[dbohning@boro-41 test]$ ls -l
-rw-r--r-- 1 dbohning dbohning 0 Jan 19 22:01 file
lrwxrwxrwx 1 dbohning dbohning 4 Jan 19 22:01 link -> file
$ sudo chown --no-dereference root:root link
$ ls -l
-rw-r--r-- 1 dbohning dbohning 0 Jan 19 22:01 file
lrwxrwxrwx 1 root     root     4 Jan 19 22:01 link -> file
```

Reply:
you're right, I was thinking of mode, not ownership.

Reply:
yea i guess we need a separate dfs_lchown/lchmod()?
probably a different PR

Reply:
Either a separate `lchown` or `flags` to pass `O_NOFOLLOW`, similar to some of the `dfs_sys` functions. (Ashley is right about mode/chmod - it's not changeable for symlinks). I remember we had a similar conversation before and decided adding `l*` (link) variants of functions would introduce a lot of clutter. FWIW, `dfs_sys_setattr` allows passing `O_NOFOLLOW`, so datamover tools could call that.
I think a different PR is okay if we don't want to add `flags` to `dfs_chmod`.

Reply:
ok a flag like dfs_sys provides is probably a good idea. i'll refresh the PR tomorrow

Reply:
Thanks!